src/index.js

```diff
@@ -133,15 +133,18 @@ class ReactTooltip extends React.Component {
     this.bindWindowEvents(resizeHide) // Bind global event for static method
   }
 
-  componentWillReceiveProps (props) {
-    const { ariaProps } = this.state
-    const newAriaProps = parseAria(props)
-
+  static getDerivedStateFromProps (nextProps, prevState) {
+    const { ariaProps } = prevState
+    const newAriaProps = parseAria(nextProps)
     const isChanged = Object.keys(newAriaProps).some(props => {
```

Comment:
Not within the scope of this PR, but it seems like this check doesn't cover the case where `newAriaProps` is a subset of the previous `ariaProps`?

E.g.,

```ts
// prevProps
a = { a: 1, b: 2, c: 3 }

// new props
b = { a: 1, b: 2 }

// returns false, but `a` and `b` actually have different props
const isChanged = Object.keys(b).some(prop => b[prop] !== a[prop])
```

So maybe adding a length check would do?

E.g.,

```ts
const isChanged = Object.keys(newAriaProps).some(props => {
  return newAriaProps[props] !== ariaProps[props]
}) && Object.keys(newAriaProps).length === Object.keys(ariaProps)
```